packages/ra-no-code/src/ui/ImportResourceDialog.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { FormEvent, useState } from 'react';
+import { parse } from 'papaparse';
+import {
+    Button,
+    Dialog,
+    DialogProps,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    RootRef,
+    TextField,
+} from '@material-ui/core';
+import { useDropzone } from 'react-dropzone';
+import {
+    useResourceConfigurations,
+    getFieldDefinitionsFromRecords,
+} from '../ResourceConfiguration';
+import { Record, useDataProvider, useRefresh } from 'ra-core';
+import { useHistory } from 'react-router-dom';
+
+export const ImportResourceDialog = (props: ImportResourceDialogProps) => {
+    const dataProvider = useDataProvider();
+    const [parsing, setParsing] = useState(false);
+    const [file, setFile] = useState<File>();
+    const [resource, setResource] = useState<string>();
+    const { addResource } = useResourceConfigurations();
+    const history = useHistory();
+    const refresh = useRefresh();
+
+    const onDrop = (acceptedFiles: File[]) => {
+        if (acceptedFiles.length > 0) {
+            const acceptedFile = acceptedFiles[0];
+            if (acceptedFile) {
+                setFile(acceptedFile);
+                setResource(acceptedFile.name.split('.')[0]);
+            }
+        }
+    };
+
+    const handleClose = () => {
+        if (props.onClose) {
+            props.onClose();
+        }
+    };
+
+    const handleSubmit = (event: FormEvent) => {
+        event.preventDefault();
+
+        if (resource && file) {
+            setParsing(true);
+            parse<Record>(file, {
+                header: true,
+                complete: ({ data }) => {
+                    data.forEach(record => {
+                        if (record.id) {
+                            dataProvider.create(resource, { data: record });
+                        }
+                    });
+                    setParsing(false);
+                    const fields = getFieldDefinitionsFromRecords(data);
+                    addResource({ name: resource, fields });
+                    history.push(`/${resource}`);
+                    handleClose();
+                    refresh();
```

Comment:
Why would you need that?

Reply:
When you import multiple files for the same resource, it ensure the list is updated with the new records

Reply:
Made it clearer in the comments and used only when needed